csrc/fusion_profiler.cpp

```diff
@@ -0,0 +1,737 @@
+           << " " << std::setw(9) << std::setprecision(3) << fp.compile_time_ms
+           << " " << std::setw(9) << std::setprecision(3) << fp.kernel_time_ms
+           << " " << std::setw(11) << std::setprecision(2)
+           << fp.effective_bandwidth_gbs << " " << std::setw(9)
+           << std::setprecision(2) << fp.percentage_peak_bandwidth;
+        first_prof = false;
+      } else {
+        os << std::setfill(' ') << std::right << std::fixed << std::setw(5)
+           << "-"
+           << " " << std::setw(5) << "-"
+           << " " << std::setw(11) << "-"
+           << " " << std::setw(9) << "-"
+           << " " << std::setw(9) << "-"
+           << " " << std::setw(9) << "-"
+           << " " << std::setw(11) << "-"
+           << " " << std::setw(9) << "-";
+      }
+      std::stringstream grid;
+      grid << "[" << std::get<0>(kp.grid) << ", " << std::get<1>(kp.grid)
+           << ", " << std::get<2>(kp.grid) << "]";
+      std::stringstream block;
+      block << "[" << std::get<0>(kp.block) << ", " << std::get<1>(kp.block)
+            << ", " << std::get<2>(kp.block) << "]";
+      std::stringstream cluster;
+      cluster << "[" << std::get<0>(kp.cluster) << ", "
+              << std::get<1>(kp.cluster) << ", " << std::get<2>(kp.cluster)
+              << "]";
+      std::stringstream smem;
+      smem << "[" << kp.dynamic_shared_mem << ", " << kp.static_shared_mem
+           << "]";
+      os << std::setfill(' ') << std::right << std::fixed << " " << std::setw(4)
+         << idx << " " << std::setw(10) << kp.name << " " << std::setw(11)
+         << std::setprecision(3) << kp.time_ms;
+
+      if (fp.verbose) {
+        os << " " << std::setw(11) << std::setprecision(3)
+           << kp.compile_time_ms;
+      }
+
+      os << " " << std::setw(13) << std::setprecision(2)
+         << kp.effective_bandwidth_gbs << " " << std::setw(9)
+         << std::setprecision(2) << kp.percentage_peak_bandwidth << " "
+         << std::setw(9) << std::setprecision(3)
+         << ((double)kp.input_bytes / 1000000.0) << " " << std::setw(9)
+         << std::setprecision(3) << ((double)kp.output_bytes / 1000000.0) << " "
+         << std::setw(16) << smem.str() << " " << std::setw(6) << kp.registers
+         << " " << std::setw(16) << grid.str() << " " << std::setw(16)
+         << block.str();
+      if (fp.verbose) {
+        os << " " << std::setw(16) << cluster.str() << " " << std::setw(5)
+           << kp.device << " " << std::setw(5) << kp.stream << " "
+           << std::setw(20) << kp.device_name << " " << std::setw(14)
+           << std::setprecision(2) << kp.peak_bandwidth_gbs;
+      }
+      os << std::endl;
+      ++idx;
+    }
+  }
+  return os;
+}
+
+std::mutex FusionProfiler::singleton_lock_;
+FusionProfiler* FusionProfiler::singleton_ = nullptr;
+
+FusionProfiler::FusionProfiler()
+    : cupti_disabled_(false),
+      cupti_buffer_((size_t)4 * 1024),
```

Comment:
Replace by `BUF_SIZE`?

Reply:
I decided to use a `static constexpr` for a data member under `FusionProfiler` to handle the `BUF_SIZE` consistently.